### ISSUE TITLE
[Feature] Look for "data-test-" attributes in Codegen

### ISSUE THREAD

**Issue by e-oz** [microsoft/playwright#20713]
By default, codegen uses too wide selectors, such us "getByText" - there might be many elements with the same text on the page (if text is short), so generated tests will pretty often throw errors about not precise selectors.

Please consider prioritising data-test- attributes: `data-test-id`, `data-test-name`, and any other "data-test-" attributes. They are  added exactly for tests and, therefore, should be used by test tools when present.

**Comment by mxschmitt** [comment]
We already prioritize `data-testid` by default and treat it as the highest scoring. You can set a different test id attribute via `playwright.selectors.setTestIdAttribute('..')`.

Would that work for you?

**Comment by e-oz** [comment]
And, data-test-id is not enough. Not every element is unique on the page, and sometimes we need data-test-name. 
Please, consider prioritising data-test-* attributes.

**Comment by mxschmitt** [comment]
> In my config I have testIdAttribute: 'data-test-id',

It depends how you invoke it, if you invoke codegen via the CLI, then the testIdAttribute does not get used. You need to invoke the codegen via the VSCode extension.

> I've tried with data-testid (although it was difficult to withstand that ugliness) - Codegen doesn't see such attributes.

Could you share how your DOM looks like or which site you are testing against? This would help us understand why its not working in your scenario.

> And, data-test-id is not enough. Not every element is unique on the page, and sometimes we need data-test-name.
Please, consider prioritising data-test-* attributes.

We can reformulate the feature request in "supporting multiple test-ids" - that would work.

**Comment by e-oz** [comment]
> You need to invoke the codegen via the VSCode extension.

No, thanks, I use a different IDE ;) It's kind of sad that Codegen has this limitation.

> Could you share how your DOM looks like or which site you are testing against? This would help us understand why its not working in your scenario.

The code is not public, but I can share a screenshot, where you can see that element has `data-testid` and Playwright Codegen doesn't see it:

<img width="1312" alt="Screenshot 2023-02-07 at 15 00 12" src="https://user-images.githubusercontent.com/526352/217266612-d5282224-aeb2-4114-9f13-f1a6e2197148.png">

> We can reformulate the feature request in "supporting multiple test-ids" - that would work.

The point of this feature request is to use better selectors to make Codegen usable, because with getByText() generated tests throw so many errors, that our testers just ignore Codegen's existence - it takes more time to fix the errors in generated tests than create tests manually.

If "supporting multiple test-ids" means that there will be some way to make Codegen prioritize "data-test-name" or "data-test-value" attributes (and not be dependent on VSCode extensions), then please reformulate. You have all the power to reject or edit this feature request :)

**Comment by mxschmitt** [comment]
If you use a different IDE, you can actually anywhere inside your test put `await page.pause()` and then click `Pick Locator` which will also launch the codegen at the position. This will reflect the `testIdAttribute` from the config.

Changed the title of the feature request.

**Comment by e-oz** [comment]
I've tried that:
```ts
import test from '@playwright/test';

test.only('example', async ({page}) => {
  await page.pause();
});
```
and it just reported:
```
 4 passed (16.3s)
```
(4 browsers are set in config).
<img width="1026" alt="Screenshot 2023-02-07 at 17 18 59" src="https://user-images.githubusercontent.com/526352/217301765-2d85333f-9e7c-4249-a1da-80686a1ddeb3.png">

You know, your hostility against feature requests is not so pleasant, so maybe I will just try some other tool.
Let me help you with this one

**Comment by e-oz** [comment]
And limit number of workers, and set only one browser... too much dancing for such basic functionality.
My attempt to make PW Codegen useful has failed, I don't care anymore.

**Comment by mtunell** [comment]
> It depends how you invoke it, if you invoke codegen via the CLI, then the testIdAttribute does not get used. You need to invoke the codegen via the VSCode extension.

Just commenting here for others since I saw this comment when googling and went "aawh too bad", it works now, not sure when it got implemented though. e.g. `playwright codegen --target=python --test-id-attribute=data-test-id` works great, it picks it up and generates page.get_by_test_id() for `data-test-id` instead of the default `data-testid`
